src/internals/ir/callGraph.ts

```diff
@@ -139,82 +182,234 @@ export class CallGraph {
+          : `receiver_${func.id}`;
       default:
         unreachable(func);
     }
   }
 
   /**
-   * Processes an expression, identifying static and method calls to add edges.
-   * @param expr The AST expression to process.
-   * @param callerId The ID of the calling node.
+   * Analyzes the AST for function calls and adds edges between caller and callee nodes.
+   * Additionally, sets flags on nodes based on their properties (e.g., if they call 'send').
+   * @param astStore The AST store to analyze.
    */
-  private processExpression(expr: AstExpression, callerId: CGNodeId) {
-    if (expr.kind === "static_call") {
-      const staticCall = expr as AstStaticCall;
-      const functionName = staticCall.function?.text;
+  private analyzeFunctionCalls(astStore: TactASTStore) {
+    for (const entry of astStore.getProgramEntries()) {
+      if (entry.kind === "contract") {
+        const contract = entry as AstContract;
+        const contractName = contract.name.text;
+        for (const declaration of contract.declarations) {
+          if (
+            declaration.kind === "function_def" ||
+            declaration.kind === "contract_init" ||
+            declaration.kind === "receiver"
+          ) {
+            const func = declaration as
+              | AstFunctionDef
+              | AstContractInit
+              | AstReceiver;
+            const funcNodeId = this.astIdToNodeId.get(func.id);
+            if (funcNodeId !== undefined) {
+              let callsSend = false;
+              forEachExpression(func, (expr) => {
+                this.processExpression(expr, funcNodeId, contractName);
+                if (this.isSendCall(expr)) {
+                  callsSend = true;
+                }
+              });
+              // Set the flag if the function calls 'send'
+              if (callsSend) {
+                const funcNode = this.getNode(funcNodeId);
+                if (funcNode) {
+                  funcNode.setFlag(FLAG_CALLS_SEND);
+                }
+              }
+            }
+          }
+        }
+      } else if (entry.kind === "function_def") {
+        const func = entry as AstFunctionDef;
+        const funcNodeId = this.astIdToNodeId.get(func.id);
+        if (funcNodeId !== undefined) {
+          let callsSend = false;
+          forEachExpression(func, (expr) => {
+            this.processExpression(expr, funcNodeId);
+            if (this.isSendCall(expr)) {
+              callsSend = true;
+            }
+          });
+          if (callsSend) {
+            const funcNode = this.getNode(funcNodeId);
+            if (funcNode) {
+              funcNode.setFlag(FLAG_CALLS_SEND);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Determines if the given expression is a 'send' call.
+   * @param expr The expression to check.
+   * @returns True if the expression is a 'send' call; otherwise, false.
+   */
+  private isSendCall(expr: AstExpression): boolean {
```

Comment:
I would move this to `src/internals/tact/util.ts` in order to make it available for other lints.

Reply:
move it